### PR TITLE
[12.0][NEW] website_sale_tax_toggle: Display price with or without taxes by toggle button in website shop

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT24533
Please @carlosdauden @Tardo @rafaelbn Can you review??

For testing in runbot remember assign any tax to products

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please don't use plurals in module name.

**Review by HaraldPanten — APPROVED** [review]
Hi @sergio-teruel this module looks really good. Functional review 👍 

Just a micro-recommendation (non blocking, for sure)... Wouldn't you prefer to set a more specific message for included/excluded taxes? IMO something like "**Show** prices with taxes" or "**Show** prices with taxes **included**", might improve a little bit the UX. What do you think?

**Comment by sergio-teruel** [comment]
@pedrobaeza @HaraldPanten Changes done!!!!

**Comment by HaraldPanten** [comment]
@sergio-teruel it still looking good to me. Just another picky comment.

As you know, normally clients are very demanding in all front-end related stuff. As a recommendation, I would consider adding extra space between "Sort by" and "Show prices with taxes includes". 

**What about something like:**


Captura de pantalla 2020-07-08 a las 15 03 33


**Instead of:**


Captura de pantalla 2020-07-08 a las 15 07 11


**I think it'll look better in mobile as well. This is the actual appearance:**


Screenshot_20200708-150832

File: website_sale_tax_toggle/models/res_users.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            taxed = request.session.get(
+                'tax_toggle_taxed',
+                self.env.user.with_context(
+                    skip_tax_toggle_check=True
+                ).has_group(tax_included)
+            )
+            if taxed:
+                return True if group_ext_id != tax_excluded else False
+            else:
+                return True if group_ext_id != tax_included else False
```
Comment (Tardo) [inline website_sale_tax_toggle/models/res_users.py]:
```suggestion
            return group_ext_id != (tax_excluded if taxed else tax_included)
```

File: website_sale_tax_toggle/static/src/js/website_sale_tax_toggle.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            ev.preventDefault();
+            var self = this;
+            var $data = $(ev.currentTarget).parents(".js_tax_toggle_management:first");
+            this._rpc({
+                route: $data.data('controller'),
+            })
+            .done(function (result) {
+                $data.find('input').prop("checked", result);
+                window.location.reload();
+            })
```
Comment (Tardo) [inline website_sale_tax_toggle/static/src/js/website_sale_tax_toggle.js]: Perhaps can be good add some feedback if something fails
```suggestion
            });
```

**Review by Tardo — APPROVED** [review]
LGTM :+1: Only non-blocking comments.

**Comment by mamcode** [comment]
> As a recommendation, I would consider adding extra space between "Sort by" and "Show prices with taxes includes".

+1

**Review by mamcode — APPROVED** [review]
LGTM, thanks.

**Comment by sergio-teruel** [comment]
@HaraldPanten @Tardo @mamcode Changes done!!

**Comment by HaraldPanten** [comment]
Looks perfect. In mobile as well 👍

**Review by rafaelbn — APPROVED** [review]
Functional review :+1:

**Comment by rafaelbn** [comment]
website_sale_tax_toggle

**Pull request merged**